### ISSUE TITLE
Test: throwinfinallyintryfilter3.exe fails in Linux/ARM

### ISSUE THREAD

**Issue by myungjoo** [dotnet/coreclr#4775]
CoreCLR Unit Test `throwinfinallyintryfilter3` (`/JIT/jit64/eh/basics/throwinfinallyintryfilter3`) fails in Linux/ARM.

```
bash-3.2# ./corerun throwinfinallyintryfilter3.exe 
in try
  in try
    in try
      in try
      in finally
    in filter
in filter
  in finally
in filter

Unhandled Exception: System.Exception: Exception of type 'System.Exception' was thrown.
   at func(Int32 i)
Aborted (core dumped)
```

1. Before diving into exception handling code of CoreCLR, it'd be much appreciated if I could get confirmed on the validity of the test case, `throwinfinallyintryfilter3`.

Using a .cs file that shows the same exception behavior (I'm not fluent in IL-ASM, so I created one), it shows the same exception:
```
bash-3.2# ./corerun csc.exe /lib:. /r:mscorlib.dll throw2.cs
Microsoft (R) Visual C# Compiler version 42.42.42.42
Copyright (C) Microsoft Corporation. All rights reserved.

throw2.cs(6,31): warning CS0168: The variable 'e' is declared but never used
throw2.cs(19,31): warning CS0168: The variable 'e' is declared but never used
bash-3.2# ./corerun throw2.exe
in filter...19
in filter...6
What The Heck
in filter...19

Unhandled Exception: System.Exception: Nah..
   at ConsoleApplication.Program.func()
Aborted (core dumped)
bash-3.2# cat throw2.cs
namespace ConsoleApplication { public class Program {

  public static void Main(string[] args) {
    try {
      func();
    } catch (System.Exception e) when (decide(10006, "6")) {
      System.Console.WriteLine("Handled.");
    }
  }
  public static void func() {
    try {
      try {
        throw new System.Exception("Throw Inner");
      } finally {
         System.Console.WriteLine("What The Heck");
         throw new System.Exception("Nah..");
      }
      throw new System.Exception("Throw");
    } catch (System.Exception e)  when ( decide(19, "19") ) {
      System.Console.WriteLine("Hello World!");
    }
  }
  public static bool decide(int x, string str) {
    System.Console.Write("in filter...");
    System.Console.WriteLine(str);
    if (x > 10000)
      return true;
    else
      return false;
  }

} }
```

Now, it seems that generating an exception during handling the exception disables exception handling of the secondary exception in CoreCLR. Is this normal? or is this a problem as `throwinfinallytryfilter3.exe` suggests? (`thowinfinallytryfilter1/2.exe` works and returns 100)

**Comment by janvorli** [comment]
The exception "Nah..." should be handled in the catch in Main, so it is an exception handling problem.
Could you please try remove the filter from the catch in Main and see how it behaves? The strange thing in your log is that the filter 6 is not called the 2nd time. Running your test on the desktop CLR gives:
in filter...19
in filter...6
What The Heck
in filter...19
**in filter...6**
Handled.

**Comment by myungjoo** [comment]
Sure, I will.

**Comment by myungjoo** [comment]
Uh.. the fix for #4779 also fixes this. As PR for #4779 is being ready. I'm closing this.


```
bash-3.2# ./corerun throw2.exe 
in filter...19
in filter...6
What The Heck
in filter...19
in filter...6
Handled.
bash-3.2# ./corerun throwinfinallyintryfilter3.exe 
in try
  in try
    in try
      in try
      in finally
    in filter
in filter
  in finally
in filter
caught an exception!
bash-3.2# echo $?
100
```

**Comment by myungjoo** [comment]
The issue appeared to be resolved by a side-effect of another patch that was tested with the fix for #4779. This requires a fix from @parjong. 

@parjong, Please submit your commit that you've shown this morning and make PR via this issue.

**Comment by parjong** [comment]
@myungjoo That patch is submitted as PR #4857
